### PR TITLE
Clarify the docs for MObject.animate, MObject.set and Variable.

### PR DESCRIPTION
**Pull request by BorisTheBrave**
## Overview: What does this pull request change?
Change various docstrings

## Motivation and Explanation: Why and how do your changes improve the library?
I think the docs were unclear. 

## Links to added or changed documentation pages
https://manimce--2407.org.readthedocs.build/en/2407/reference/manim.mobject.mobject.Mobject.html
https://manimce--2407.org.readthedocs.build/en/2407/reference/manim.mobject.numbers.Variable.html

## Further Information and Comments
None

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/mobject/mobject.py [hunk 1]
```diff
@@ -225,7 +225,17 @@ def construct(self):
 
     @property
     def animate(self):
-        """Used to animate the application of a method.
+        """Used to animate the application of any method of :code:`self`.
+
+        Any method called on :code:`animate` is converted to an animation that repeatedly calls that method on the mobject.
+
+        For example, :code:`square.set_fill(WHITE)` sets the fill color of a square, while :code:`sqaure.animate.set_fill(WHITE)` animates this action.
```
Comment (BorisTheBrave) [inline manim/mobject/mobject.py:232]: This line is copied from the introductory tutorial, I think it gives important context about how animate is meant to be used.

File: manim/mobject/mobject.py [hunk 2]
```diff
@@ -225,7 +225,17 @@ def construct(self):
...
+
+        Multiple methods can be put in a single animation once via chaining:
```
Comment (BorisTheBrave) [inline manim/mobject/mobject.py:234]: This feature was previously only mentioned in a warning. Warnings should be supplementary to documentation, not hiding extra features.

File: manim/mobject/mobject.py [hunk 3]
```diff
@@ -511,7 +519,9 @@ def __isub__(self, other):
     def set(self, **kwargs) -> "Mobject":
         """Sets attributes.
 
-        Mainly to be used along with :attr:`animate` to
+        I.e. ``my_mobject.set(foo=1)`` applies ``my_mobject.foo = 1``.
```
Comment (BorisTheBrave) [inline manim/mobject/mobject.py:522]: Like with animate I felt an inline example was necessary as the method signature doesn't give anything away.

File: manim/mobject/numbers.py [hunk 4]
```diff
@@ -286,17 +286,14 @@ def get_value(self):
 
 
 class Variable(VMobject, metaclass=ConvertToOpenGL):
```
Comment (BorisTheBrave) [inline manim/mobject/numbers.py:288]: Personally I think Variable is a bad class, but I've done my best to fix the confusions it was causing me.

I've removed references to "python variable". The class doesn't relate to python variables at all - the value is actually stored in a ValueTracker.

File: manim/mobject/mobject.py (hunk 1 above)
Comment (behackl) [reply manim/mobject/mobject.py:232]: 👍

File: manim/mobject/mobject.py (hunk 2 above)
Comment (behackl) [reply manim/mobject/mobject.py:234]: 👍

File: manim/mobject/mobject.py (hunk 3 above)
Comment (behackl) [reply manim/mobject/mobject.py:522]: 👍

File: manim/mobject/mobject.py [hunk 5]
```diff
@@ -225,7 +225,17 @@ def construct(self):
 
     @property
     def animate(self):
-        """Used to animate the application of a method.
+        """Used to animate the application of any method of :code:`self`.
+
+        Any method called on :code:`animate` is converted to an animation of applying that method on the mobject itself.
+
+        For example, :code:`square.set_fill(WHITE)` sets the fill color of a square, while :code:`sqaure.animate.set_fill(WHITE)` animates this action.
```
Comment (behackl) [inline manim/mobject/mobject.py:232]: These two lines are a bit long, please break them:

```suggestion
        Any method called on :code:`animate` is converted to an animation of applying
        that method on the mobject itself.

        For example, :code:`square.set_fill(WHITE)` sets the fill color of a square,
        while :code:`sqaure.animate.set_fill(WHITE)` animates this action.
```

File: manim/mobject/numbers.py (hunk 4 above)
Comment (behackl) [reply manim/mobject/numbers.py:288]: I like your proposed changes a lot, thank you. I agree that `Variable` is an excellent candidate for being renamed.

File: manim/mobject/numbers.py [hunk 6]
```diff
@@ -286,17 +286,14 @@ def get_value(self):
...
-    """A class for displaying text that continuously updates to reflect the value of a python variable.
-
-    Automatically adds the text for the label and the value when instantiated and added to the screen.
+    """A class for displaying text that shows "label = value" with the value continuously updated from a :class:`~.ValueTracker`.
```
Comment (behackl) [inline manim/mobject/numbers.py:289]: Line length.

```suggestion
    """A class for displaying text that shows "label = value" with
    the value continuously updated from a :class:`~.ValueTracker`.
```

**Review by behackl — APPROVED** [review]
Thank you for your contribution! Left two minor suggestions regarding line length, please take a look -- but other than that, this can be merged.

**Comment by BorisTheBrave** [comment]
@behackl I've applied your suggestions, please merge when you are ready.

**Pull request merged**